### PR TITLE
feat: barometric (b) key handling

### LINKED ISSUE #6786: [ISIS] pressing B changes ALT while in STD mode
### Aircraft Version

Experimental

### Build info

```json
{
  "built": "2022-02-22T10:29:25+00:00",
  "ref": "experimental",
  "sha": "c736b13b89efac7b7e819d1d4e010a350f777a0d",
  "actor": "aguther",
  "event_name": "manual",
  "pretty_release_name": "experimental:c736b13b"
}
```

### Describe the bug

I like to use the B key, to quickly set altimeters. Today I thought about, wether its possible to "preset" the QNH while on STD, so it would match the second I turn back to QNH. So i pressed B in cruise and I noticed the ALT on the ISIS changed. I changed the ISIS to QNH (which was still at previous setting) and back to STD and the ALT was correct again. Hitting B again changes the ALT again.

I guess the STD mode also has its own altimeter setting which you can change, because QNH mode had a different setting.

### Expected behavior

Nothing should change, when on STD.

### Steps to reproduce

1. Set QNH to something other than current QNH
2. Change to STD
3. Press B key -> ALT changes
4. Change to QNH (shows old QNH from 1.)

### References (optional)

IRL experience

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### PR DESCRIPTION
**Pull request by tracernz**
Fixes #6786

## Summary of Changes

Handle the 'B' key ourselves, and set QNH/STD depending on the transition altitude and transition level entered in the MCDU, defaulting to 18000/FL180. This matches behaviour of default avionics like the CJ4, and improves the usefulness of the key for non-USA simmers. Also fixes the ISIS so that pressing the B key while in STD mode doesn't change the underlying baro correction value to a QNH.

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Set up a short flight plan and ensure transition altitude is set on PERF TO, and transition level on PERF APPR. Check that pressing the B key on the takeoff runway sets the local QNH, press it during climb prior to trans alt and make sure it continues to set local QNH... press it after trans alt and ensure the altimeter switches to STD. Press at various points to make sure it stays STD. During descent press before and after trans level and ensure it changes to local QNH below trans level and keep trying to ensure the same behaviour at various points.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/systems/extras-host/index.tsx
Comment (frankkopp) [inline fbw-a32nx/src/systems/extras-host/index.tsx:53]: Is there a specific reason you prefer to avoid the construcor?

I'd at least want to keep the log output so finding issues is easier.

File: fbw-a32nx/src/systems/extras-host/modules/baro_handler/BaroHandler.ts
Comment (frankkopp) [inline fbw-a32nx/src/systems/extras-host/modules/baro_handler/BaroHandler.ts:46]: Might warrant a console output (log, warning) - not fully clear to me if this is a kind of error condition or not.

**Review by frankkopp — CHANGES_REQUESTED** [review]
You could have hooked this into the KeyInterceptor and done all the logic in a separate class. Why did you create another class intercepting keys?
I think it would be cleaner to keep these in one class. Of course then the Ctrl+E stuff could be moved out as well to be consistent.

File: fbw-a32nx/src/systems/extras-host/modules/baro_handler/BaroHandler.ts
Comment (tracernz) [reply fbw-a32nx/src/systems/extras-host/modules/baro_handler/BaroHandler.ts:46]: Just means all the ADRs are off or failed. Not a code error.

**Comment by tracernz** [comment]
> You could have hooked this into the KeyInterceptor and done all the logic in a separate class. Why did you create another class intercepting keys? I think it would be cleaner to keep these in one class. Of course then the Ctrl+E stuff could be moved out as well to be consistent.

I don't think that's a good idea at all. I'd rather have each separate function encapsulated in it's own class rather than mixing a whole bunch of different things into a messy file full of unrelated stuff. The KeyInterceptManager (KeyEventManager with framework 0.4) is designed for exactly this.

**Comment by tracernz** [comment]
> You could have hooked this into the KeyInterceptor and done all the logic in a separate class. Why did you create another class intercepting keys?
I think it would be cleaner to keep these in one class. Of course then the Ctrl+E stuff could be moved out as well to be consistent.

This is the style I am used to using with the framework in my day job. As you're the maintainer of this code, I'll defer to you, and put the intercept in KeyInterceptor.

**Review by frankkopp — APPROVED** [review]

**Review by 2hwk — APPROVED** [review]

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #7902
Tier of Testing  : 1
Date             : 29/03/2023

Testing Process: 
Set up a short flight plan and ensure transition altitude is set on PERF TO, and transition level on PERF APPR. Check that pressing the B key on the takeoff runway sets the local QNH, press it during climb prior to trans alt and make sure it continues to set local QNH... press it after trans alt and ensure the altimeter switches to STD. Press at various points to make sure it stays STD. During descent press before and after trans level and ensure it changes to local QNH below trans level and keep trying to ensure the same behaviour at various points. - **All OK**

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

File: .github/CHANGELOG.md
Comment (tracernz) [inline .github/CHANGELOG.md:82]: Changelog 0.12

**Pull request closed without merging**
